internal/protocoltest/jsonrpc/deserializers.go

```diff
@@ -0,0 +1,133 @@
+// Code generated by smithy-go-codegen DO NOT EDIT.
+package jsonrpc
+
+import (
+	"context"
+	"github.com/awslabs/smithy-go/middleware"
+)
+
+type awsAwsjson11_deserializeOpEmptyOperation struct {
+}
+
+func (*awsAwsjson11_deserializeOpEmptyOperation) ID() string {
+	return "awsAwsjson11_deserializeOpEmptyOperation"
+}
+
+func (m *awsAwsjson11_deserializeOpEmptyOperation) HandleDeserialize(ctx context.Context, in middleware.DeserializeInput, next middleware.DeserializeHandler) (
+	out middleware.DeserializeOutput, metadata middleware.Metadata, err error,
+) {
+	out, metadata, err = next.HandleDeserialize(ctx, in)
+	if err != nil {
+		return out, metadata, err
+	}
+
+	output := &EmptyOperationOutput{}
+	out.Result = output
+
+	return out, metadata, err
+}
+
+type awsAwsjson11_deserializeOpJsonEnums struct {
+}
+
+func (*awsAwsjson11_deserializeOpJsonEnums) ID() string {
+	return "awsAwsjson11_deserializeOpJsonEnums"
+}
+
+func (m *awsAwsjson11_deserializeOpJsonEnums) HandleDeserialize(ctx context.Context, in middleware.DeserializeInput, next middleware.DeserializeHandler) (
+	out middleware.DeserializeOutput, metadata middleware.Metadata, err error,
+) {
+	out, metadata, err = next.HandleDeserialize(ctx, in)
+	if err != nil {
+		return out, metadata, err
+	}
+
+	output := &JsonEnumsOutput{}
+	out.Result = output
+
+	return out, metadata, err
+}
```

Comment:
It doesn't look like the shape deserialization functions are being generated. Is that Intentional?

Reply:
Yes, this is just serialization. There's a basic framework for deser so that it'll build and the protocol tests won't panic.